querydsl-sql/src/main/java/com/querydsl/sql/SQLSerializer.java

```diff
@@ -471,7 +471,19 @@ void serializeForDelete(QueryMetadata metadata, RelationalPath<?> entity) {
         dmlWithSchema = false;
 
         if (metadata.getWhere() != null) {
-            append(templates.getWhere()).handle(metadata.getWhere());
+            serializeForWhere(metadata);
+        }
+    }
+
+    void serializeForWhere(QueryMetadata metadata) {
+        boolean requireSchemaInWhere = templates.isRequiresSchemaInWhere();
+        if (requireSchemaInWhere) {
+            dmlWithSchema = true;
+        }
+        append(templates.getWhere()).handle(metadata.getWhere());
+
+        if (requireSchemaInWhere) {
+            dmlWithSchema = false;
```

Comment:
Makes sense. Should I change the other methods to do the same?

`serializeForDelete`, `serializeForMerge`,`serializeForInsert` and `serializeForUpdate`